# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaStreamTrack.cpp
# Commit: 243a33803d16
# Full Hash: 243a33803d16d50cfcc8cf0a5c4739a5ed08dc55
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2018-11-23 21:37:23
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1423241 - Refactor DecodedStream. r=jya
#   
#   This removes DecodedStream's use of MediaStreamListener in favor of
#   MediaStreamTrackListener. This change has however rippled through to a lot
#   more cleanup, per below.
# ==============================================================================

diff -r 12f91d8a1f69 -r 243a33803d16 dom/media/MediaStreamTrack.cpp
--- a/dom/media/MediaStreamTrack.cpp	Fri Nov 23 15:01:48 2018 +0000
+++ b/dom/media/MediaStreamTrack.cpp	Fri Nov 23 15:02:03 2018 +0000
@@ -430,10 +430,9 @@
 
 already_AddRefed<MediaStreamTrack> MediaStreamTrack::Clone() {
   // MediaStreamTracks are currently governed by streams, so we need a dummy
-  // DOMMediaStream to own our track clone. The dummy will never see any
-  // dynamically created tracks (no input stream) so no need for a SourceGetter.
+  // DOMMediaStream to own our track clone.
   RefPtr<DOMMediaStream> newStream =
-      new DOMMediaStream(mOwningStream->GetParentObject(), nullptr);
+      new DOMMediaStream(mOwningStream->GetParentObject());
 
   MediaStreamGraph* graph = Graph();
   newStream->InitOwnedStreamCommon(graph);